synapse/handlers/room.py

```diff
@@ -1286,6 +1241,30 @@ async def create_event(
             )
             events_to_send.append((encryption_event, encryption_context))
 
+        if "name" in room_config:
+            name = room_config["name"]
+            name_event, name_context = await create_event(
+                EventTypes.Name,
+                {"name": name},
+                True,
+                room_id=room_id,
+                sender=creator.user.to_string(),
+                state_key="",
```

Comment:
Why do we need to provide the `room_id` and `sender` here but not for other events we've created? (Same question for `topic` below.)